examples/localstack.rs

```diff
@@ -14,19 +14,19 @@ async fn main() -> Result<(), s3::Error> {
 
     let host_ip = container.get_host().await;
     let host_port = container.get_host_port_ipv4(4566).await;
-
     // Set up AWS client
     let endpoint_url = format!("http://{host_ip}:{host_port}");
-    let region_provider = RegionProviderChain::default_provider().or_else("us-east-1");
     let creds = s3::config::Credentials::new("fake", "fake", None, None, "test");
-    let config = aws_config::defaults(BehaviorVersion::v2023_11_09())
-        .region(region_provider)
+
+    let config = aws_sdk_s3::config::Builder::default()
+        .behavior_version(BehaviorVersion::v2024_03_28())
```

Comment:
This one, in fact - not required change, because depends on your `Cargo.lock` (e.g - I have `aws-smithy-runtime 1.1.6` in it - where `v2024_03_28` doesn't present, and it corresponds to minimal versions declared in `Cargo.toml`). 

But let's keep it, since it's more fresh version